src/main/java/com/direwolf20/buildinggadgets/common/building/Region.java

```diff
@@ -0,0 +1,240 @@
+package com.direwolf20.buildinggadgets.common.building;
+
+import com.direwolf20.buildinggadgets.common.building.placement.IPlacementSequence;
+import com.google.common.collect.AbstractIterator;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3i;
+import net.minecraft.world.gen.structure.StructureBoundingBox;
+
+import java.util.Iterator;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * Represents a region in the world with a finite and nonzero size.
+ * <p>Javadoc are copied from {@link AxisAlignedBB} with some modifications.</p>
+ */
+public final class Region extends StructureBoundingBox implements IPlacementSequence {
```

Comment:
Could you maybe not make this subclass StructurBoundingBox and instead make truly immutable?